internal/core/command/get_test.go

```diff
@@ -108,9 +116,145 @@ func TestNewGetCommandNoCorrelationIDInContext(t *testing.T) {
 
 func TestNewGetCommandInvalidBaseUrl(t *testing.T) {
 	device := testDevice
+	req := newRequestWithHeaders(map[string]string{clients.ContentType: clients.ContentTypeCBOR}, http.MethodGet)
 	device.Service.Addressable.Address = "!@#$"
-	_, err := NewGetCommand(device, testCommand, "", context.Background(), nil, logger.NewMockClient())
+	_, err := NewGetCommand(device, testCommand, context.Background(), nil, logger.NewMockClient(), req)
 	if err == nil {
 		t.Errorf("The invalid URL error was not properly propagated to the caller")
 	}
 }
+
+func TestNewGetCommandContentType(t *testing.T) {
+	tests := []struct {
+		name             string
+		requestedHeaders []string
+		expectedHeaders  map[string]string
+		req              *http.Request
+	}{
+		{
+			"cbor content type header propagated",
+			[]string{clients.ContentType},
+			map[string]string{
+				clients.ContentType: clients.ContentTypeCBOR},
+			newRequestWithHeaders(map[string]string{
+				clients.ContentType: clients.ContentTypeCBOR}, http.MethodGet),
+		},
+		{
+			"json content type header propagated",
+			[]string{clients.ContentType},
+			map[string]string{
+				clients.ContentType: clients.ContentTypeJSON},
+			newRequestWithHeaders(map[string]string{
+				clients.ContentType: clients.ContentTypeJSON}, http.MethodGet),
+		},
+		{
+			"no content type header provided",
+			[]string{clients.ContentType},
+			map[string]string{
+				clients.ContentType: ""},
+			newRequestWithHeaders(map[string]string{
+				clients.ContentType: ""}, http.MethodGet),
+		},
+		{
+			"user-agent header not propagated, if-modified-since not propagated, max-forwards not propagated",
+			[]string{"User-Agent", "If-Modified-Since", "Max-Forwards"},
+			map[string]string{
+				"User-Agent":        "",
+				"If-Modified-Since": "",
+				"Max-Forwards":      "",
```

Comment:
I think my last comment discouraging the use commonly used headers was not communicated properly. Can we avoid using common HTTP headers and instead use some irrelevant string? The reason for this is that the functionality which propagates headers is done in a way which very easily allows to add or remove headers in the future and if we decide to add one that is explicitly not expected in the tests we will have a lot of test break, which means low maintainability. You can effectively achieve the same functional test coverage by using a "random" header name, like `NonPropegatedHeader`, so that way we will unlikely cause issues in the future and it is clear the intent of the test. Please apply this comment to the rest of the test which specify headers like this including the tests in `put_test.go`